### PR TITLE
Rewrite DELETE docs

### PR DESCRIPTION
**Pull request by justindeguzman**
### Changelog category (leave one):
- Documentation (changelog entry is not required)

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit 757fa750e63c0a96aadf8b38b9b987ec68a42983 with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/55856/757fa750e63c0a96aadf8b38b9b987ec68a42983/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://github.com/ClickHouse/ClickHouse/blob/master/.github/PULL_REQUEST_TEMPLATE.md?plain=1">❌ failure</a></td></tr>
<tbody>
</table>

**Review by CurtizJ — APPROVED** [review]

**Review by thomoco — COMMENTED** [review]
Very nice set of changes, thank you

I was wondering - should we perhaps be recommending wide parts in all use cases where many deletions are expected?

File: docs/en/sql-reference/statements/delete.md
Comment (evillique) [inline docs/en/sql-reference/statements/delete.md:16]: Should we specifically say that this is a "lazy" type of deletion and not that the deletion will happen asynchronously in the background, because we will not do anything to actively delete the data but will simply wait until the next merge happens? As a result, the deleted data can be stored for a long time without actually being deleted.

File: docs/en/sql-reference/statements/delete.md
Comment (evillique) [inline docs/en/sql-reference/statements/delete.md:18]: I am also not sure about this statement because yes, it won't really affect the performance as much as it will just take an undefined amount of time to delete the data.

File: docs/en/sql-reference/statements/delete.md
Comment (evillique) [inline docs/en/sql-reference/statements/delete.md:29]: What does `re-create the table` mean in this context? TRUNCATE TABLE deletes all the data in the table but it does not DROP it, so there isn't really anything to recreate

File: docs/en/sql-reference/statements/delete.md
Comment (evillique) [inline docs/en/sql-reference/statements/delete.md:31]:
```suggestion
If you anticipate frequent deletes, consider using a [custom partitioning key](/en/engines/table-engines/mergetree-family/custom-partitioning-key). You can then use the [`ALTER TABLE...DROP PARTITION`](/en/sql-reference/statements/alter/partition#drop-partitionpart) command to quickly drop all rows associated with that partition.
```

File: docs/en/sql-reference/statements/delete.md
Comment (evillique) [inline docs/en/sql-reference/statements/delete.md:37]: This is not true, the DELETE (marking the rows as deleted) is synchronous by default (mutations_sync = 2), and there is currently nothing except `OPTIMIZE TABLE ... FINAL` that can force ClickHouse to actually synchronously delete the data.

**Review by evillique — CHANGES_REQUESTED** [review]
We should also compare it to the `ALTER TABLE ... DELETE` in ways that `ALTER DELETE` will actually delete the data, but it will recreate all the affected parts, which may take a lot of resources at the moment of deletion. At the same time, the `DELETE` query will not load ClickHouse as much because we only mark the rows as deleted, but it may take a lot of time for the data to be actually deleted.

File: docs/en/sql-reference/statements/delete.md
Comment (justindeguzman) [reply docs/en/sql-reference/statements/delete.md:18]: What's a better way to phrase this?

**Deleting large volumes of data frequently can negatively affect ClickHouse performance.**

Or should we just omit this?

File: docs/en/sql-reference/statements/delete.md
Comment (evillique) [inline docs/en/sql-reference/statements/delete.md:16]: This would be more precise
```suggestion
Deleted rows are internally marked as deleted immediately and will be automatically filtered out of all subsequent queries. Cleanup of data happens during the next merge. As a result, it is possible that for an unspecified period, data is not actually deleted from storage and is only marked as deleted.
```

File: docs/en/sql-reference/statements/delete.md
Comment (evillique) [reply docs/en/sql-reference/statements/delete.md:20]: Maybe something like this?
```
Deleting large volumes of data with the DELETE statement can negatively affect SELECT query performance.
```

File: docs/en/sql-reference/statements/delete.md
Comment (evillique) [inline docs/en/sql-reference/statements/delete.md:22]: Why were all mentions of `DELETE` being lightweight deleted?

Comment (justindeguzman) [reply docs/en/sql-reference/statements/delete.md:22]: Sorry, I'm not sure I understand this comment. Can you expand further?

File: docs/en/sql-reference/statements/delete.md
Comment (evillique) [inline docs/en/sql-reference/statements/delete.md:17]: Should we also somewhere at the top of the page mention and refer people to a "heavy" `ALTER table DELETE` query that will actually give you guarantees (asynchronously or synchronously) of deleting your data in a predictable period of time?

File: docs/en/sql-reference/statements/delete.md
Comment (evillique) [reply docs/en/sql-reference/statements/delete.md:22]: I mean previously DELETEs in the text were explicitly called lightweight, now it is just DELETEs (except line 75)

**Comment by justindeguzman** [comment]
@evillique Thank you for reviewing. I believe I've addressed all of your feedback above in the most recent commit.

Please let me know if there are any more changes I should make.

**Comment by den-crane** [comment]
> Very nice set of changes, thank you
> 
> I was wondering - should we perhaps be recommending wide parts in all use cases where many deletions are expected?

May be it has sense to put _row_exists into a separate file **always**, even if a part is compact. Compact parts are introduced to solve the overhead of small inserts, but delete happens when inserts already done and this column does not exists in the data.bin.

**Review by evillique — APPROVED** [review]
Looks perfect, thank you!

**Comment by nellicus** [comment]
thank you @evillique @justindeguzman ! 🙏

**Pull request merged**
